docs/DATA-AUTHORIZATION.md

```diff
@@ -0,0 +1,58 @@
+# Data Authorization
+
+## Meadowlark Authorization Modes
+
+There are four authorization modes, described in the following sections.
+
+### Ownership
+
+Assign "vendor" to the roles for this default authorization model when creating the API client.
+
+"Whoever creates a record has access to it". ClientA submits a POST request with a Student. ClientB guesses the document ID for that student and issues a GET request for it: access is denied. ClientA issues the same GET request: access granted, 200 response.
+
+This does not affect validation: if ClientB submits a StudentEducationOrganizationAssociation with "ClientA's student", then the validation passes despite the fact that ClientB "cannot directly see" the student.
+
+### Host Full Access
+
+Assign "host" to the roles for this default authorization model when creating the API client.
+
+This mode is intended for API hosting providers so that they can run synchronization processes, with access to read all documents unfiltered.
+
+### Admin
+
+Assign "admin" to the roles for this default authorization model when creating the API client.
+
+This mode is specific to the internal OAuth2 provider and client management API, allowing the API client to create and manage other API clients.
+
+### Assessment
+
+Assign "assessment" to the roles for this default authorization model when creating the API client.
+
+(warning) This is not a unique authorization model, and it should be used in addition to "vendor". This role allows the API client to bypass the usual referential integrity checks when issuing a POST or PUT request.
```

Comment:
I'd like to avoid using the term "referential integrity" since that strongly implies an underlying relational db implementation with foreign key constraints. The Meadowlark term is "reference validation".

Reply:
You raise a good point. I was probably the originally writer of that sentence. Generally speaking then, we need to shift language across all tools to talk about "reference validation" instead of "referential integrity".